### PR TITLE
Borg emag tweaks

### PR DESCRIPTION
**Pull request by LankLTE**
## About the PR

When emagged, borgs now use the below lawset (syndicate lawset). Additionally borgs are stunned for 5 seconds when emagged.

image

## Why / Balance

Laws are as per discussion with CE- the new lawset helps mitigate the issue of borgs immediately outing their emag-ee before any orders can be given, while also freeing borgs of their laws if not given explicity orders (which is more fun, I feel). The stun is to give people time to give the borgs orders, while also making it way more obvious that you've been emagged and discouraging doing it in the open (though I doubt anyone was doing that anyways)

## Technical details

Emagged borgs are now given an updated lawset, defined in the EmagSiliconLaw component. Law 0 is static since it needs to be hardcoded to use the uid, unfortunately I don't know a way around that. After borgs are emagged they're stunned for 5 seconds if they haven't been emagged already (to prevent perma-stuns).

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: Lank
- tweak: The Syndicate EMAG now provides cyborgs with an updated lawset on use, making typical laws only apply to the user and preventing your new "friend" from outing your activities, in addition to stunning the cyborg for a short time.

### CODE REVIEW

**Comment by JoeHammad1844** [comment]
oh my based

**Comment by Anzarot121** [comment]
How will this work with a revenant?

**Comment by LankLTE** [comment]
> How will this work with a revenant?

It’ll work as it does currently- the Law 0 will apply to the revenant, but they can only emag a Borg that has the panel open.

File: Content.Server/Silicons/Laws/SiliconLawSystem.cs [hunk 1]
```diff
@@ -146,6 +157,9 @@ protected override void OnGotEmagged(EntityUid uid, EmagSiliconLawComponent comp
         base.OnGotEmagged(uid, component, ref args);
         NotifyLawsChanged(uid);
         EnsureEmaggedRole(uid, component);
+
+        if(!HasComp<EmaggedComponent>(uid))
+            _stunSystem.TryParalyze(uid, TimeSpan.FromSeconds(5), true);
```
Comment (EmoGarbage404) [inline Content.Server/Silicons/Laws/SiliconLawSystem.cs:162]: just put the timespan on the comp

**Comment by deltanedas** [comment]
new meta you can do over binary channel as pai: ask a borg to state laws and watch it take 5 years

**Comment by LankLTE** [comment]
> new meta you can do over binary channel as pai: ask a borg to state laws and watch it take 5 years

Or watch as it ignores you because Borgs don’t have to listen to pAI

File: Content.Server/Silicons/Laws/SiliconLawSystem.cs (hunk 1 above)
Comment (LankLTE) [reply Content.Server/Silicons/Laws/SiliconLawSystem.cs:162]: ok

**Comment by LankLTE** [comment]
Stun time is now in the component

**Comment by LankLTE** [comment]
I don’t have even the slightest idea what that test fail is I’ll work on it tomorrow

**Comment by LankLTE** [comment]
Good now

File: Content.Server/Silicons/Laws/SiliconLawSystem.cs [hunk 2]
```diff
@@ -146,6 +157,9 @@ protected override void OnGotEmagged(EntityUid uid, EmagSiliconLawComponent comp
         base.OnGotEmagged(uid, component, ref args);
         NotifyLawsChanged(uid);
         EnsureEmaggedRole(uid, component);
+
+        if(!HasComp<EmaggedComponent>(uid) && component.StunTime != 0)
```
Comment (EmoGarbage404) [inline Content.Server/Silicons/Laws/SiliconLawSystem.cs:161]: you don't need this condition

File: Content.Shared/Silicons/Laws/Components/EmagSiliconLawComponent.cs [hunk 3]
```diff
@@ -22,6 +23,18 @@ public sealed class EmagSiliconLawComponent : Component
...
+    /// The laws that the borg is given when emagged. 
+    /// </summary>
+    [DataField("emagLaws", required: true, customTypeSerializer: typeof(PrototypeIdListSerializer<SiliconLawPrototype>))]
+    public List<string> EmagLaws = new();
+
+    /// <summary>
+    /// How long the borg is stunned when it's emagged. Setting to 0 will disable it. 
+    /// </summary>
+    [DataField("stunTime"), ViewVariables(VVAccess.ReadWrite)]
+    public int StunTime = 0;
```
Comment (EmoGarbage404) [inline Content.Shared/Silicons/Laws/Components/EmagSiliconLawComponent.cs:36]: just make this a timespan (it'll Just Work:tm: with the yaml)

**Review by EmoGarbage404 — APPROVED** [review]

File: Content.Server/Silicons/Laws/SiliconLawSystem.cs (hunk 2 above)
Comment (LankLTE) [reply Content.Server/Silicons/Laws/SiliconLawSystem.cs:161]: Can borgs not be emagged more than once? I was worried we'd end up with like, perma stunning borgs or something and figured it was safer to keep it

Comment (deltanedas) [reply Content.Server/Silicons/Laws/SiliconLawSystem.cs:161]: it already checks if EmaggedComponent exists before raising the emag event

**Comment by LankLTE** [comment]
I think test fails are unrelated

**Pull request merged**